src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -671,11 +681,17 @@ def __write_status_file(self):
             status_file_payload['status']['substatus'].append(self.__metadata_for_healthstore_substatus_json)
         if self.__configure_patching_substatus_json is not None:
             status_file_payload['status']['substatus'].append(self.__configure_patching_substatus_json)
-        if os.path.isdir(self.status_file_path):
+        if os.path.isdir(self.complete_status_file_path):
             self.composite_logger.log_error("Core state file path returned a directory. Attempting to reset.")
-            shutil.rmtree(self.status_file_path)
+            shutil.rmtree(self.complete_status_file_path)
+
+        # Write complete status file <seq.no>.complete
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(json.dumps(status_file_payload)), mode='w+')
 
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.status_file_path, '[{0}]'.format(json.dumps(status_file_payload)), mode='w+')
+        complete_status_file = self.__read_complete_status_file_raw_data(self.complete_status_file_path)
```

Comment:
There are no actions between <>.complete.status write and <>.status writes. We should not need this read again then. Can directly use status_file_payload to write to <>.status